examples/VirtualMachine/VirtualMachine.Tests.ps1

```diff
@@ -1,76 +1,96 @@
 BeforeAll {
   Import-Module Az.InfrastructureTesting
+
+  $Script:rgName = 'rg-test'
+  $Script:vmName = 'simpleLinuxVM1'
+  $Script:location = 'westus3'
 }
 
-Describe 'Verify Virtual Machine Exists' {
-  it 'Should contain a Virtual Machine with the given name' {
+Describe 'Verify Virtual Machine' {
+  It 'Should contain a Virtual Machine with the given name - Confirm-AzBPResource' {
+    #arrange
+    $params = @{
+      ResourceType = "VirtualMachine"
+      ResourceGroupName = $rgName
+      ResourceName = $vmName
+    }
+
+    #act
+    $result = Confirm-AzBPResource @params
+
+    #assert
+    $result.Success | Should -Be $true
+  }
+
+  It "Should contain a Virtual Machine named $vmName - ConfirmAzBPResource" {
     #arrange
-    $rgName = 'rg-test'
-    $vmName = 'simpleLinuxVM1'
+    $params = @{
+      ResourceType = "VirtualMachine"
+      ResourceGroupName = $rgName
+      ResourceName = $vmName
+      PropertyKey = 'Name'
+      PropertyValue = $vmName
+    }
 
+    #act
+    $result = Confirm-AzBPResource @params
+
+    #assert
+    $result.Success | Should -Be $true
+  }
+
+  It 'Should contain a Virtual Machine with the given name' {
     #act
     $result = Confirm-AzBPVirtualMachine -ResourceGroupName $rgName -VirtualMachineName $vmName
 
     #assert
     $result.Success | Should -Be $true
   }
-}
 
-Describe 'Verify Virtual Machine Does Not Exist' {
-  it 'Should not contain a Virtual Machine with the given name' {
+  It 'Should not contain a Virtual Machine with the given name' {
     #arrange
-    $rgName = 'rg-test'
-    $vmName = 'noSimpleLinuxVM1'
+    $params = @{
+      ResourceGroupName = $rgName
+      VirtualMachineName = 'noSimpleLinuxVM1'
```

Comment:
This is also missing the BeforeAll block for that parameter, but it begs the question of if we should keep these non-existent resources hard-coded here since we only use it once. I'm okay hard coding these

Reply:
Thanks! Fixed!

Reply:
Regarding leaving the values inline: I'm usually of the opinion that inline magic numbers and magic string literals should typically be avoided. With that said, I'm not as dogmatic about it when it comes to single use string literals as I am with magic numbers. I could go either way with this one, but I tend to lean toward using a variable.